stats/statsview.cpp

```diff
@@ -1080,10 +1101,10 @@ void StatsView::plotScatter(const std::vector<dive *> &dives, const StatsVariabl
 		series->append(dive, x, y);
 
 	// y = ax + b
-	auto [a, b] = linear_regression(points);
-	if (!std::isnan(a)) {
+	struct regression_data reg = linear_regression(points);
+	if (!std::isnan(reg.a)) {
 		auto [minx, maxx] = axisX->minMax();
 		auto [miny, maxy] = axisY->minMax();
-		addLinearRegression(a, b, minx, maxx, miny, maxy, xAxis, yAxis);
+		addLinearRegression(reg.a, reg.b, reg.res2, reg.r2, minx, maxx, miny, maxy, xAxis, yAxis);
```

Comment:
Why not pass the structure instead of the four members separately?

If you don't want to export the structure in the ".h" file, you can simply forward declare it and pass a const reference.